src/components/functional/VersionSelector.jsx

```diff
@@ -31,7 +33,7 @@ class VersionSelector extends React.Component {
     componentWillReceiveProps(nextProps) {
         let rVersion = nextProps.removeVersion;
         if (rVersion !== undefined && rVersion !== this.state.lastRemovedVersion) {
-            let newValues = this.state.values.filter((v) => {
+            const newValues = this.state.values.filter((v) => {
```

Comment:
Since a version is presumably unique: Why can't you just make sure that `versions` does not contain `removeVersion` or overwrite `values` and letting the parent component handle the removal?